protocols/imap/src/main/java/org/apache/james/imap/decode/parser/SearchCommandParser.java

```diff
@@ -49,6 +50,18 @@
  * Parse SEARCH commands
  */
 public class SearchCommandParser extends AbstractUidCommandParser {
+    private static class Context {
+        private Optional<Charset> previousCharset = Optional.empty();
+
+        public void setCharset(Charset charset) {
+            previousCharset = Optional.of(charset);
+        }
+
+        public Charset getCharset() {
+            return previousCharset.orElse(null);
+        }
+    }
```

Comment:
Q: Why do we use Optional when directly returning null?

Reply:
A: because I wanted to minimize the changes in this complex call, it would have had required to change 20+ method signature.